src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -190,6 +190,14 @@ public static ValidationException validateKnnField(
             return exception;
         }
 
+        String vectorDataType = (String) fieldMap.get(VECTOR_DATA_TYPE_FIELD);
```

Comment:
can `fieldMap.get(VECTOR_DATA_TYPE_FIELD)` be null?

Reply:
if it can be null then the type casting to string will cause a NPE. Please handle this gracefully.

Reply:
It won't cause a NPE.

Reply:
why?? casting a null to string cause NPE. I am missing something here.

Reply:
I tested it and it didn't cause NPE.